docs/ref/paginator.txt

```diff
@@ -226,6 +295,75 @@ Attributes
 
     The associated :class:`Paginator` object.
 
+``AsyncPage`` class
+===================
+
+.. versionadded:: 5.2
+
+.. class:: AsyncPage(object_list, number, paginator)
+
+    Asynchronous version of :class:`Page`.
+
+    AsyncPage does not behave like a sequence.
+
+    You need to use :meth:`.AsyncPage.alength`. and :meth:`.AsyncPage.agetitem`.
+
+Methods
+-------
+
+.. method:: AsyncPage.agetitem()
+
+    Returns the point at the specified index
+    for :attr:`.AsyncPage.object_list`.
+
+.. method:: AsyncPage.alength()
+
+    Returns length of :attr:`.AsyncPage.object_list`.
+
+.. method:: AsyncPage.ahas_next()
+
+    See :meth:`.Page.has_next`.
+
+.. method:: AsyncPage.ahas_previous()
+
+    See :meth:`.Page.has_previous`.
+
+.. method:: AsyncPage.ahas_other_pages()
+
+    See :meth:`.Page.has_other_pages`.
+
+.. method:: AsyncPage.anext_page_number()
+
+    See :meth:`.Page.next_page_number`.
+
+.. method:: AsyncPage.aprevious_page_number()
+
+    See :meth:`.Page.previous_page_number`.
+
+.. method:: AsyncPage.astart_index()
+
+    See :meth:`.Page.start_index`.
+
+.. method:: AsyncPage.aend_index()
+
+    See :meth:`.Page.end_index`.
+
+
+Attributes
+----------
+
+.. attribute:: AsyncPage.object_list
+
+    See :attr:`.Page.object_list`.
+
+.. attribute:: AsyncPage.number
+
+    See :attr:`.Page.number`.
+
+.. attribute:: AsyncPage.paginator
+
+    See :attr:`.Page.paginator`.
```

Comment:
```suggestion
    ``AsyncPage`` has the same attributes and signatures as :class:`Page`, as
    well as asynchronous versions of all the same methods, using an ``a``
    prefix - for example, use ``await async_page.ahas_next()`` rather than
    ``page.has_next()``.

    However, as ``AsyncPage`` does not act like a sequence of
    ``AsyncPage.object_list`` when using ``len()`` or indexing it directly,
    instead the asynchronous methods :meth:`.AsyncPage.alength` and
    :meth:`.AsyncPage.agetitem` should be used.

    ``AsyncPage`` has the following additional methods:

    .. method:: agetitem()

        Returns the point at the specified index for ``AsyncPage.object_list``.

    .. method:: alength()

        Returns length of ``AsyncPage.object_list``.
```